### ISSUE TITLE
bug: font-loading isn't synchronizing between Next.js application and Storybook

### ISSUE THREAD

**Issue by neoncitylights** [elliniaresearch/app#98]
### Problem Description

Starting with #90 / 76c28c0, this caused a regression in Storybook's font loading but fixed Next'js font loading. This is sort of slightly tangled since the font issue relates to Storybook, TailwindCSS, and Next.js.

## How font loading works
First, how font loading currently works right now:
 - Next.js loads fonts using `next/font` which supports loading fonts from Google Fonts, or locally. This package optimizes font loading by optimizing the font files and avoiding network requests by then self-hosting the file at compile time.
 - We declare the fonts in `src/app/layout.tsx`, which is the global layout that Next.js uses for all pages in routing. The relevant code includes the following (some code purposely excluded):
    ```tsx
    import { Inter, JetBrains_Mono } from 'next/font/google' // where we import
    import { twMerge } from 'tailwind-merge'
    
    // where we configure our fonts
    const inter = Inter({
	    subsets: ['latin'],
	    display: 'swap',
	    variable: '--font-inter',
    })
    export const jbMono = JetBrains_Mono({
	    subsets: ['latin'],
	    display: 'swap',
	    variable: '--font-jetbrains-mono',
    })
  
    export default function RootLayout({
	    children,
    }: {
	    children: React.ReactNode
    }) {
            // where we interpolate our font variable names into the <html> class attribute
	    return (
              <html lang="en" className={twMerge(inter.variable, jbMono.variable)}>
    ```
  - In Tailwind's configuration file, `tailwind.config.js` (at the root), we customize the values for some of Tailwind's default classnames specifically related to fonts/typefaces. See:
```
    fontFamily: {
      sans: ['var(--font-inter)'],
      mono: ['var(--font-jetbrains-mono)'],
    }
```
  - We declare our global CSS file at `src/app/globals.css`, which includes the following below. This gets imported into `src/app/layout.tsx`, which is one of the few files where global CSS is allowed.
    ```css
    @tailwind base;
    @tailwind components;
    @tailwind utilities;
    
    body {
	    @apply font-sans;
    }
    ```
  - We also import this file into `.storybook/preview.ts`. Since tailwind has custom CSS declarative rules (like `@tailwind`/`@apply`), we configure PostCSS which compiles and normalizes the CSS. PostCSS still clearly works, because this is part of the captured and compiled output when you view Storybook and inspect element: 
<img width="402" alt="Screenshot of Storybook's CSS for the html element" src="https://github.com/wandering-app/app/assets/4723983/c5aff53a-77e3-4e46-86ad-bc11878b3477">

However, because Storybook loads components in an `<iframe>`, it is thus a different, nested [browsing context](https://developer.mozilla.org/en-US/docs/Glossary/Browsing_context). And above, the `--font-inter` is where we say it's grayed out, because the variable is not set.

## Regression
Previously, (highlighted in orange/red) Next.js did not load the Inter font, and because CSS supports font family fallbacks, it fell back to loading the **default sans-serif** font installed on a user's OS. But, Storybook loaded the Inter font.

However, 76c28c0 fixed the font-loading for the Next.js side, and broke Storybook. So technically, it's been subtly broken by being unsynchronized between both applications!

### Steps to Reproduce

 - Run `npm run dev` (starts compiling and running Next.js application)
 - Run `npm run storybook`

Compare both side-by-side. Storybook fails to load the font.

**Comment by neoncitylights** [comment]
One thing we could do to resolve, but I believe is not ideal, is creating a `.storybook/preview-head.html`, and loading fonts there. The reason this isn't ideal is because the development environment should match stage environment and production environment as close as possible, and this will cause some (slight) maintenance burden by having to synchronize the sans typeface and monospace typeface for Storybook with the typefaces configured for Next.js.

See: https://storybook.js.org/docs/react/configure/story-rendering for information about injecting HTML into Storybook's `<head>` with the `preview-head.html` file

**Comment by neoncitylights** [comment]
Temporarily fixed in #99. Will investigate soon into a more permanent fix
